### PR TITLE
Total test coverage 4657

### LINKED ISSUE #14: introduce 100% coverage to be on par with connect
We decided that 100% test coverage is valuable to the team, that's why we introduced

https://github.com/SUSE/connect/blob/master/spec/spec_helper.rb#L8

Here we should consider the same

### PR DESCRIPTION
**Pull request by ikapelyukhin**
Will resolve #14.

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
@thutterer will review your pull request :dancers: check https://trello.com/c/5HtnB9ZV/4657-rmt-100-test-coverage
  @thutterer: Please review this pull request using our guidelines:
  * test for acceptance criteria / functionality
  * check if the new code is covered with tests
  * check for unintended consequences
  * encourage usage of the boyscout rule
  * make sure the code is architected in the best way
  * check that no unnecessary technical debt got introduced
  * make sure that no unnecessary FIXMEs or TODOs got added

File: spec/fixtures/files/dummy_repo/repodata/abf421e45af5cd686f050bab3d2a98e0a60d1b5ca3b07c86cb948fc1abfa675e-primary.xml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  <summary>Apples dummy RPM</summary>
+  <description>Dummy apples RPM</description>
+  <packager></packager>
+  <url></url>
+  <time file="1500453272" build="1500453253"/>
+  <size package="1934" installed="14" archive="268"/>
+<location href="apples-0.1-0.x86_64.rpm"/>
+  <format>
+    <rpm:license>Public</rpm:license>
+    <rpm:vendor>obs://build.suse.de/Devel:SCC:rmt-dummy</rpm:vendor>
```
Comment (thutterer) [inline spec/fixtures/files/dummy_repo/repodata/abf421e45af5cd686f050bab3d2a98e0a60d1b5ca3b07c86cb948fc1abfa675e-primary.xml]: Do we want/need to have this info in a public repo?

File: spec/requests/api/connect/v3/systems/activations_controller_spec.rb [hunk 2]
```diff
@@ -19,6 +22,13 @@
       its(:code) { is_expected.to eq '401' }
     end
 
+    context 'when not credentials are wrong' do
```
Comment (thutterer) [inline spec/requests/api/connect/v3/systems/activations_controller_spec.rb]: ~~not~~

File: spec/requests/api/connect/v3/systems/products_controller_spec.rb [hunk 3]
```diff
@@ -20,19 +27,30 @@
...
 
-      let(:payload) do
+      let(:product) { FactoryGirl.create(:product, :with_not_mirrored_repositories) }
+      let(:error_json) do
         {
-          identifier: product_not_mirrored.identifier,
-          version: product_not_mirrored.version,
-          arch: product_not_mirrored.arch
-        }
+          'type' => 'error',
```
Comment (thutterer) [inline spec/requests/api/connect/v3/systems/products_controller_spec.rb]: nitpick: could be done with symbols instead string+rocket, right?

File: spec/requests/api/connect/v3/systems/products_controller_spec.rb [hunk 4]
```diff
@@ -20,19 +27,30 @@
...
+      subject { response }
+
+      let(:base_product) { FactoryGirl.create(:product, :with_not_mirrored_repositories, :with_mirrored_extensions) }
+      let(:product) { base_product.extensions[0] }
       let(:error_json) do
         {
           'type' => 'error',
-          'error' => "Not all mandatory repositories are mirrored for product #{product_not_mirrored.friendly_name}",
-          'localized_error' => "Not all mandatory repositories are mirrored for product #{product_not_mirrored.friendly_name}"
+          'error' => "Unmet product dependencies, activate one of these products first: #{base_product.friendly_name}",
```
Comment (thutterer) [inline spec/requests/api/connect/v3/systems/products_controller_spec.rb]: :rocket: :scream:

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -1,4 +1,6 @@
 require 'simplecov'
+
+SimpleCov.minimum_coverage 100
```
Comment (thutterer) [inline spec/spec_helper.rb]: :+1:

File: spec/fixtures/files/dummy_repo/repodata/abf421e45af5cd686f050bab3d2a98e0a60d1b5ca3b07c86cb948fc1abfa675e-primary.xml (hunk 1 above)
Comment (ikapelyukhin) [reply spec/fixtures/files/dummy_repo/repodata/abf421e45af5cd686f050bab3d2a98e0a60d1b5ca3b07c86cb948fc1abfa675e-primary.xml]: Yeah, sure -- it's just a fixture for specs. That repo with fixture files was built on OBS, but we don't care where it was build for anything RMT-related.

File: spec/requests/api/connect/v3/systems/products_controller_spec.rb (hunk 4 above)
Comment (ikapelyukhin) [reply spec/requests/api/connect/v3/systems/products_controller_spec.rb]: :fire::fire::fire::radioactive::rocket::radioactive::fire::fire::fire:

**Pull request merged**